packages/@aws-cdk/sdk-v2-to-v3-adapter/lib/coerce-api-parameters.ts

```diff
@@ -0,0 +1,54 @@
+import { UINT8ARRAY_PARAMETERS } from './parameter-types';
+
+type ApiParameters = { [param: string]: any };
+
+/**
+ * Given a minimal AWS SDKv3 call definition (service, action, parameters),
+ * coerces nested parameter values into a Uint8Array if that's what the SDKv3 expects.
+ */
+export function coerceApiParametersToUint8Array(service: string, action: string, parameters: ApiParameters = {}): ApiParameters {
+  const pathsToCoerce = UINT8ARRAY_PARAMETERS?.[service.toLowerCase()]?.[action.toLowerCase()] ?? [];
+  for (const path of pathsToCoerce) {
+    coerceToUint8Array(parameters, path.split('.'));
+  }
+  return parameters;
+}
+
+/**
+ * Given an object and a path, traverse to the leaf and coerce the value into a Uint8Array.
+ * This function mutates the original object.
+ *
+ * @example
+ *    const obj = { a: { b: { c: '1' } } }
+ *    coerceToUint8Array(obj, ['a', 'b', 'c'])
+ *    assert(obj == {a: {b: {c: new Uint8Array([49]} } })
+ */
+export function coerceToUint8Array(obj: any, path: string[]): any {
+  const currentKey = path?.[0];
+  if (!currentKey) {
+    return;
+  }
+
+  // recurse into array
+  if (currentKey === '*' && Array.isArray(obj)) {
+    obj.map(e => coerceToUint8Array(e, path.slice(1)));
```

Comment:
This doesn't work if the array contains the coercible values directly (instead of the array containing objects that have the coercible values). 

You'll need to `return` the new map.